src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ComService.kt

```diff
@@ -172,4 +183,77 @@ class ComService(
         if (it.direction == SearchDirection.ASC) "asc" else "desc",
       )
     }
+
+  private fun PrisonerSearchPrisoner.isUnstartedRecordEligibleForCvl(): Boolean {
+    // Person is parole eligible - not eligible for CVL
+    if (this.paroleEligibilityDate != null) {
+      // Parole eligibility - must be set and in the future
+      if (this.paroleEligibilityDate.isAfter(LocalDate.now())) {
+        return false
+      }
+    }
+
+    // Person is dead - not eligible for CVL
+    if (this.legalStatus == "DEAD") {
+      return false
+    }
+
+    // Person is on an indeterminate sentence - not eligible for CVL
+    if (this.indeterminateSentence) {
+      return false
+    }
+
+    // Person does not have a CRD - not eligible for CVL
+    if (this.conditionalReleaseDate == null) {
+      return false
+    }
+
+    // Person is on an ineligible extended determinate sentence - not eligible for CVL
+    if (this.confirmedReleaseDate != null) {
+      // PED cannot be in the future
+      if (this.paroleEligibilityDate != null) {
+        if (this.paroleEligibilityDate.isAfter(LocalDate.now())) {
+          return false
+        }
+      }
+
+      val dateStart = this.conditionalReleaseDate.minusDays(4)
+      // ARD must be between CRD - 4 days and CRD - if outside this then not eligible
+      if (!(this.confirmedReleaseDate.isAfter(dateStart) && this.confirmedReleaseDate.isBefore(this.conditionalReleaseDate))) {
+        return false
+      }
+
+      // APD with a PED in the past means they were successful parole applicant on a later attempt
+      if (actualParoleDate != null) {
+        return false
+      }
+    }
+
+    // Person has a status other than ACTIVE x or INACTIVE TRN - not eligible for CVL
+    if (this.status != null) {
+      if (!(this.status.startsWith("ACTIVE") || this.status == "INACTIVE TRN")) {
+        return false
+      }
+    }
+
+    // Person's ARD or CRD is in the past - not eligible for CVL
+    if (this.confirmedReleaseDate != null) {
+      if (this.confirmedReleaseDate.isBefore(LocalDate.now()) || this.conditionalReleaseDate.isBefore(LocalDate.now())) {
+        return false
+      }
+    }
+
+    // Person has no CRD and a post recall release date - Recall case - not eligible for CVL
+    if (this.confirmedReleaseDate != null && this.conditionalReleaseDate != null && this.postRecallReleaseDate != null) {
+      return false
+    }
+
+    // Person is approved for HDC - not eligible for CVL
+    val hdcStatus = prisonApiClient.getHdcStatus(this.bookingId).block()
```

Comment:
mmh.. that's a hefty call to make for each licence.. good to have it at the end, maybe there's a bulk endpoint we can use instead?